conans/client/toolchain/cmake.py

```diff
@@ -0,0 +1,300 @@
+        {% if build_type %}
+        set(CMAKE_BUILD_TYPE "{{ build_type }}" CACHE STRING "Choose the type of build." FORCE)
+        {% endif %}
+
+        # --  - CMake.flags --> CMakeDefinitionsBuilder::get_definitions
+        {%- for it, value in definitions.items() %}
+        {%- if it.startswith('CONAN_') %}
+        set({{ it }} "{{ value }}")
+        {%- else %}
+        set({{ it }} "{{ value }}" CACHE STRING "Value assigned from the Conan toolchain" FORCE)
+        {%- endif %}
+        {%- endfor %}
+
+        get_property( _CMAKE_IN_TRY_COMPILE GLOBAL PROPERTY IN_TRY_COMPILE )
+        if(_CMAKE_IN_TRY_COMPILE)
+            message(STATUS "Running toolchain IN_TRY_COMPILE")
+            return()
+        endif()
+
+        message("Using Conan toolchain through ${CMAKE_TOOLCHAIN_FILE}.")
+
+        if(CMAKE_VERSION VERSION_LESS "3.15")
+            message(WARNING
+                " CMake version less than 3.15 doesn't support CMAKE_PROJECT_INCLUDE variable\\n"
+                " used by Conan toolchain to work. In order to get the same behavior you will\\n"
+                " need to manually include the generated file after your 'project()' call in the\\n"
+                " main CMakeLists.txt file:\\n"
+                " \\n"
+                "     project(YourProject C CXX)\\n"
+                "     include(\\"\\${CMAKE_BINARY_DIR}/conan_project_include.cmake\\")\\n"
+                " \\n"
+                " This file contains some definitions and extra adjustments that depend on\\n"
+                " the build_type and it cannot be done in the toolchain.")
+        else()
+            # Will be executed after the 'project()' command
+            set(CMAKE_PROJECT_INCLUDE "{{ conan_project_include_cmake }}")
+        endif()
+
+        # We are going to adjust automagically many things as requested by Conan
+        #   these are the things done by 'conan_basic_setup()'
+         # To support the cmake_find_package generators:
+        {% if cmake_module_path %}
+        set(CMAKE_MODULE_PATH {{ cmake_module_path }} ${CMAKE_MODULE_PATH}){% endif%}
+        {% if cmake_prefix_path %}
+        set(CMAKE_PREFIX_PATH {{ cmake_prefix_path }} ${CMAKE_PREFIX_PATH}){% endif%}
+
+        {% if fpic %}
+        message(STATUS "Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)")
+        set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+        {% endif %}
+
+        {% if set_rpath %}conan_set_rpath(){% endif %}
+        {% if set_std %}conan_set_std(){% endif %}
+        {% if set_libcxx %}conan_set_libcxx(){% endif %}
+
+        set(CMAKE_CXX_FLAGS_INIT "${CONAN_CXX_FLAGS}" CACHE STRING "" FORCE)
+        set(CMAKE_C_FLAGS_INIT "${CONAN_C_FLAGS}" CACHE STRING "" FORCE)
+        set(CMAKE_SHARED_LINKER_FLAGS_INIT "${CONAN_SHARED_LINKER_FLAGS}" CACHE STRING "" FORCE)
+        set(CMAKE_EXE_LINKER_FLAGS_INIT "${CONAN_EXE_LINKER_FLAGS}" CACHE STRING "" FORCE)
+    """)
```

Comment:
Would it be an option to inject additionally a current CMake toolchain to the generated one by using `include(my_toolchain.cmake)`?

Reply:
Yes, definitely. This is a very barebones initial proposal, but this would be a necessary feature.

In reality, it is possible that if you are providing your own toolchain, like the own from Android, the conan one is not very used. Some of the things that this toolchain does for cmake:

- Skip Rpath in OSX
- Define the MSVC runtime (MT, MD) based on Conan settings
- Define the libcxx based on Conan settings
- Define the cppstd based on Conan settings.

It is likely that some of these are irrelevant or unused. But yes, having a way to inject another toolchain is easy and useful.